File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetOrgUnitsUseCase.java
package org.eyeseetea.malariacare.domain.usecase;

import org.eyeseetea.malariacare.domain.boundary.repositories.IOrgUnitRepository;
import org.eyeseetea.malariacare.domain.entity.OrgUnit;

import java.util.List;

public class GetOrgUnitsUseCase {
    private IOrgUnitRepository orgUnitRepository;

    public GetOrgUnitsUseCase(IOrgUnitRepository orgUnitRepository){
        this.orgUnitRepository = orgUnitRepository;
    }

    public List<OrgUnit> execute() throws Exception {
        return orgUnitRepository.getAll();
    }
}

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetOrgUnitsUseCase.kt
package org.eyeseetea.malariacare.domain.usecase

import org.eyeseetea.malariacare.domain.boundary.repositories.IOrgUnitRepository
import org.eyeseetea.malariacare.domain.entity.OrgUnit

class GetOrgUnitsUseCase(private val orgUnitRepository: IOrgUnitRepository) {

    @Throws(Exception::class)
    fun execute(): List<OrgUnit> {
        return orgUnitRepository.getAll()
    }
}
